UnboundLocalError: local variable 'noqa' referenced before assignment

### Describe the bug

The following error occurred in sphinx when using the internationalization feature to HTML content in rst file.
Content:
```rst
.. raw:: html

  <table class="contentstable" align="center" style="margin-left: 30px"><tr>
    <td width="45%" rowspan="4">
      <img alt="Mayavi banner" src="_static/m2_about.jpg" 
            width="100%" />
    </td>
    <td width="5%" rowspan="4"></td>
    <td width="50%">
      <p class="biglink"><a class="biglink" href="application.html">
                Using the Mayavi application</a><br/>
         <span class="linkdescr">
                Understanding and using the Mayavi interactive application</span></p>
    </td></tr><tr><td width="50%">
      <p class="biglink"><a class="biglink" href="mlab.html">
        Python scripting for 3D plotting</a><br/>
         <span class="linkdescr">The simple scripting API to Mayavi</span></p>
    </td></tr><tr><td width="50%">
      <p class="biglink"><a class="biglink" href="auto/examples.html">
        Gallery and examples</a><br/>
         <span class="linkdescr">
            Example gallery of visualizations, 
            with the Python code that generates them</span></p>
    </td></tr>
  </table>
```
Error:
```bash
  File "/usr/local/lib/python3.9/site-packages/sphinx/transforms/i18n.py", line 320, in apply
    if not noqa and len(old_foot_refs) != len(new_foot_refs):
UnboundLocalError: local variable 'noqa' referenced before assignment

Exception occurred:
  File "/usr/local/lib/python3.9/site-packages/sphinx/transforms/i18n.py", line 320, in apply
    if not noqa and len(old_foot_refs) != len(new_foot_refs):
UnboundLocalError: local variable 'noqa' referenced before assignment
The full traceback has been saved in /tmp/sphinx-err-se8e55i6.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:2: all] Error 2
```

### How to Reproduce

```Dockerfile
FROM python:3.9-slim
RUN apt update; apt install -y git make
RUN git clone https://github.com/tkoyama010/mayavi-doc-translations
WORKDIR /mayavi-doc-translations
RUN git checkout 2cfddb290ca429fd720334115ba95e66c271a63a
RUN pip install -r requirements.txt
RUN make
```


### Expected behavior

Build without error.

### Your project

https://github.com/tkoyama010/mayavi-doc-translations

### Screenshots

_No response_

### OS

Ubuntu22.04

### Python version

3.9

### Sphinx version

4.5.0

### Sphinx extensions

_No response_

### Extra tools

_No response_

### Additional context

_No response_

--- a/CHANGES
+++ b/CHANGES
@@ -30,6 +30,7 @@ Bugs fixed
 * #10520: HTML Theme: Fix use of sidebar classes in ``agogo.css_t``.
 * #6679: HTML Theme: Fix inclusion of hidden toctrees in the agogo theme.
 * #10566: HTML Theme: Fix enable_search_shortcuts does not work
+* #10579: i18n: UnboundLocalError is raised on translating raw directive
 
 Testing
 --------

--- a/sphinx/transforms/i18n.py
+++ b/sphinx/transforms/i18n.py
@@ -238,6 +238,7 @@ class Locale(SphinxTransform):
                 continue  # skip if the node is already translated by phase1
 
             msgstr = catalog.gettext(msg)
+            noqa = False
 
             # See above.
             if not isinstance(node, LITERAL_TYPE_NODES):

